lib/net/nforceif/src/sys_arch.c

```diff
@@ -372,6 +313,25 @@ sys_arch_sem_wait(struct sys_sem **s, u32_t timeout)
     return 0;
 }
 
+int sys_arch_sem_try(struct sys_sem **s)
```

Comment:
Is this for a more recent lwip? I thought the `sys_arch_*` functions were the backend for lwip.
So it's a bit weird to see a new one. I wonder if this should have another prefix / none, and be made `static`.

This is also the only function to be platform dependent in this commit:
- Was the (old or new) mbox implementation taken from elsewhere in lwip?
- Is there a reference implementation we could just use?

Reply:
It's an extension to be able to implement the `sys_mbox_try*` functions. It was already static, but I added the `static` keyword to the definition now, too, to make it clear.

I don't know where the old mbox implementation came from, but the new one isn't taken from lwip, although the implementations there are probably similar (they're all system-specific ofc).

Reply:
If this is your own extension and lwip code normally doesn't have that function (including other platform specific code), then I'd recommend to rename that function; or at least a comment in the code would help, to clarify that this was added for our backend but that it isn't part of the lwip API.

Reply:
I've changed the name to `ext_sys_arch_sem_try` and added a comment.